app/src/features/surveys/view/survey-animals/animal-form-helpers.test.ts

```diff
@@ -0,0 +1,139 @@
+import { IDetailedCritterWithInternalId } from 'interfaces/useSurveyApi.interface';
+import { v4 } from 'uuid';
+import { IAnimal, IAnimalCapture, IAnimalMarking } from './animal';
+import { arrDiff, createCritterUpdatePayload, transformCritterbaseAPIResponseToForm } from './animal-form-helpers';
+
+describe('animal form helpers', () => {
+  describe('transformCritterbaseAPIResponseToForm', () => {
+    it('should return an object matching the IAnimal interface', () => {
+      const detailedResponse: IDetailedCritterWithInternalId = {
+        survey_critter_id: 1,
+        critter_id: v4(),
```

Comment:
I think it makes more sense to hardcode test values like UUIDs, so that they can be directly compared with a string literal.
```javascript
expect(result.general.critter_id).toEqual('3347d57e-033a-4a6f-9b39-893ffb590510');
```

In this case, you'd know for certain that `taxon_id` didn't get mutated inside `transformCritterbaseAPIResponseToForm`